src/main/java/org/spongepowered/common/mixin/core/world/MixinWorld.java

```diff
@@ -277,28 +294,71 @@ public boolean isBlockPowered(int x, int y, int z) {
 
         Entity entity = null;
 
-        try {
-            entity = ConstructorUtils.invokeConstructor(type.getEntityClass(), this);
-            entity.setLocation(entity.getLocation().setPosition(position));
-        } catch (Exception e) {
-            Sponge.getLogger().error(ExceptionUtils.getStackTrace(e));
+        Class<? extends Entity> entityClass = type.getEntityClass();
+        double x = position.getX();
+        double y = position.getY();
+        double z = position.getZ();
+
+        if (entityClass.isAssignableFrom(EntityPlayerMP.class) || entityClass.isAssignableFrom(EntityDragonPart.class)) {
+            // Unable to construct these
+            return Optional.absent();
+        }
+
+        net.minecraft.world.World world = (net.minecraft.world.World) (Object) this;
+
+        // Not all entities have a single World parameter as their constructor
+        if (entityClass.isAssignableFrom(EntityLightningBolt.class)) {
+            entity = (Entity) new EntityLightningBolt(world, x, y, z);
+        } else if (entityClass.isAssignableFrom(EntityEnderPearl.class)) {
+            EntityArmorStand tempEntity = new EntityArmorStand(world, x, y, z);
```

Comment:
Yes, the constructor must take an `EntityLivingBase` and I think that an armor stand is the cheapest object to construct.